src/tools/kata-ctl/src/arch/s390x/mod.rs

```diff
@@ -65,4 +66,38 @@ mod arch_specific {
     pub fn get_checks() -> Option<&'static [CheckItem<'static>]> {
         Some(CHECK_LIST)
     }
+
+    fn retrieve_cpu_facilities() -> Result<HashMap<i32, bool>> {
+        //let cpu_info = check::get_single_cpu_info(check::PROC_CPUINFO, CPUINFO_DELIMITER)?;
+        let f = std::fs::File::open(check::PROC_CPUINFO);
```

Comment:
Missing check on the file not opening correctly.

It may make sense to just make the implementations you can't test directly NOP's to allow the other architecture owners to complete the implementations. Something like the following should compile without warnings on any arch:


```rust
  #[allow(dead_code)]
  fn retrieve_cpu_facilities() -> Result<HashMap<i32, bool>> {
      unimplemented!();
  }
```